src/asyncIterableToBuffer.js

```diff
@@ -12,6 +12,11 @@ const handler = {
       console.log('New call to', key);
       obj._keys[key] = true;
     }
+    // if function, ensure context is present.
```

Comment:
This block of code belongs to a Proxy handler implementation and it will proxy obj keys. It also includes functions which has their context lost on previous implementation.

Reply:
This is something we definitely should revisit sometime but for now I just would like to ensure functions has correct context.